ci/lib/build/generate-material.sh

```diff
@@ -0,0 +1,64 @@
+#!/bin/bash
+set -o nounset -o pipefail -o errexit
+
+# full path to dir of current script
+SCRIPT_DIR=$(readlink -e $(dirname ${BASH_SOURCE[0]}))
+
+# full path to root of PF sources
+PF_SRC_DIR=$(echo ${SCRIPT_DIR} | grep -oP '.*?(?=\/ci\/)')
+
+# path to all functions
+FUNCTIONS_FILE=${PF_SRC_DIR}/ci/lib/common/functions.sh
+
+source ${FUNCTIONS_FILE}
+
+configure_and_check() {
+    CI_COMMIT_TAG=${CI_COMMIT_TAG:-}
+    CI_COMMIT_REF_SLUG=${CI_COMMIT_REF_SLUG:-}
+    SRC_FILE=${SRC_FILE:-}
+    DST_FILE=${DST_FILE:-}
+    CONTAINER_NAME=pfconfig_material
+
+    if [ -n "${CI_COMMIT_TAG}" ]; then
+        # release
+        IMAGE_TAG=${CI_COMMIT_TAG}
+    elif [ -n "${CI_COMMIT_REF_SLUG}" ]; then
+        # all branches (maintenance included and devel)
+        IMAGE_TAG=${CI_COMMIT_REF_SLUG}
+    else
+        IMAGE_TAG=localdev
+    fi
+
+    declare -p CI_COMMIT_TAG CI_COMMIT_REF_SLUG
+    declare -p IMAGE_TAG
+}
+
+run_pfconfig_container() {
+    docker run --detach --name=${CONTAINER_NAME} --rm -e PFCONFIG_PROTO=unix \
+           -e GIT_USER_NAME \
+           -e GIT_USER_MAIL \
+           -e GIT_USER_PASSWORD \
+	   -e GIT_REPO \
+	   -e CI_PIPELINE_ID \
+           -v ${PF_SRC_DIR}/conf:/usr/local/pf/conf \
+           -v ${PF_SRC_DIR}/addons/dev-helpers/bin:/usr/local/pf/addons/dev-helpers/bin \
+           -v ${PF_SRC_DIR}/ci/lib:/usr/local/pf/ci/lib \
+           -v ${PF_SRC_DIR}/config.mk:/usr/local/pf/config.mk \
+           -v ${PF_SRC_DIR}/Makefile:/usr/local/pf/Makefile \
+           ghcr.io/inverse-inc/packetfence/pfconfig:${IMAGE_TAG}
+    docker exec ${CONTAINER_NAME} /usr/bin/make material
+    docker exec ${CONTAINER_NAME} /usr/local/pf/ci/lib/release/publish-to-git.sh ${SRC_FILE} ${DST_FILE}
```

Comment:
One thing I'm not 100% sure is the usage of `--detach` in that context.

According to my tests, shell script is moving directly to `make material` command after `docker run` while `pfconfig` container is starting so I suppose in some case it could fail because `pfconfig` is not yet started. If you have better solution @julsemaan, let me know.

Reply:
I added a `sleep` statement because I hit exactly that bug while building in CI.